cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -370,6 +372,22 @@ void addApplication(Id.Application id,
   WorkflowToken getWorkflowToken(Id.Workflow workflowId, String workflowRunId);
 
   /**
+   * Add node state for the given {@link Workflow} run. This method is used to update the
+   * state of the custom actions started by Workflow.
```

Comment:
Yes. sure we can have signature similar to setStart/setStop method. Something like -
```java
updateWorkflowNodeState(ProgramId customActionId, Map<String, String> systemArgs);
updateWorkflowNodeState(ProgramId customActionId, ProgramRunStatus status, Throwable cause);
```
However for predicates we will still need different signature, since they cannot be represented by ProgramId. Let me know your thoughts.